oeps/processes/oep-0055/decisions/0001-use-backstage-to-support-maintainers.rst

```diff
@@ -0,0 +1,133 @@
+0001 Use Backstage to Support Maintainers
+#########################################
+
+Status
+******
+
+**Draft**
+
+.. TODO Upon Acceptance
+
+	* Update OEP to mention the catalog-info.yaml file instead of the readme.
+		* Relevant link https://open-edx-proposals.readthedocs.io/en/latest/processes/oep-0055-proc-project-maintainers.html#documenting-maintainers
+
+Context
+*******
+
+* :doc:`OEP-55 <../../oep-0055-proc-project-maintainers>` laid out that we should document the maintainers of a component.
+
+* As a maintainer it is hard to know what components you are maintaining.
+
+* As a community member it is currently difficult to quickly look up who the
+  maintainers are for multiple components quickly.
+
+* Given the number of repos that make up the Open edX platform and the
+  complexity of the job of maintaining them, we need a system for cataloging,
+  and making transparent the critical details of our repositories.
+
+Decision
+********
+
+To solve the above issues and to enable the potential to ease other
+discoverability and maintainership burdens, the Open edX community shall
+maintain a `backstage`_ instance and related metadata.
+
+Consequences
+************
+
+* To support the backstage instance ownership data will have to be stored in a
+  structured way.  We will follow Backstage best practices and store this data
+  in a ``catalog-info.yaml`` file in each component repository.
+
+* Once a backstage instance exists we have the opportunity to further leverage
+  it as a centralized access point for maintainers as well as community members
+  to quickly and easily assess different components and find relevant links to
+  code, documentation, owners, etc.
+
+* Backstage will also give us the ability to expose other health metrics to
+  maintainers so they can more easily assess the components they own.
+
+* As a result of adding the ``catalog-info.yaml`` file we end with two top-level
+  files that have very similar data.  If this decision is not reversed after the
+  initial pilot, it may make sense to revisit this consequence and see if we can
+  consolidate this metadata into a single file.
+
+Rejected Alternatives
+*********************
+
+Spreadsheet
+===========
+
+Pros
+----
+
+* Easy to build and get a high level view of the system.
+
+Cons
+----
+
+* Not a beautiful UI for maintainers to work with.
+
+* Data overload as we put more and more data into it with little to no control
+  over the presentation of the data.
+
+Owners in the Readme
+====================
+
```

Comment:
I don't think both should be necessarily exclusive of each other, and I think we can get the best of both world. 

As I understand the `catalog-info.yml` should have the source of truth about maintners. Thus we could make an org wide action that whenever `catalog-info.yml` is changed/updated the repo's readme would be updated/overwritten accordingly. 

There are various ways to achieve this, here is for example a very simliar purpose action: https://github.com/JasonEtco/rss-to-readme. 

The only caveate that would make this thing tricky is that some repo has an rst while other has md format... But anyway this can be overcomed by defining clear steps/requirmets for both cases about how to _make a project repo compitable with the action_.

Reply:
> We're still hashing out the layout of the README files so I think this is an open question 

Sounds reasonable, if I would like to further express my intention, where should I go? 

> A thing I've thought of before is to use something like cog to auto generate certain parts of the file from the sources of truth 

cog seems intresting!, its not exactly how I imagined it to be, but I see how it can be useful. wonder what @nedbat thinks about using it  😄

Reply:
@ghassanmas we'll have another ADR for the format/layout of the README shortly so keep in this repo for that.